saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala

```diff
@@ -125,6 +125,12 @@ object MultiModalSpRLDataModel extends DataModel {
       getTokenDistance(first, second)
   }
 
+  val isTokenAnImageConcept = property(tokens) {
+    t: Token =>
+      ((tokens(t) <~ sentenceToToken <~ documentToSentence) ~> documentToImage ~> imageToSegment)
+        .map(_.getSegmentConcept) exists (x => t.getText.contains(x))
+  }
+
```

Comment:
Yes, we need to use some kind of similarity(using wordnet or word-to-vector, ...)